apps/target_apps_store.py

```diff
@@ -32,6 +32,34 @@ def copy(self, target: FactoryClient.Target, dst_images_dir: str, dst_apps_dir:
     def apps_location(self, target: FactoryClient.Target):
         pass
 
+EXTRACTION_SCRIPT = '''#!/bin/sh
+set -e
+set -o pipefail
+if [ $# -ne 1 ] ; then
+    echo "Usage: $0 <decryption key>"
+    exit 1
+fi
+
+archive=$(grep --text --line-number '# ARCHIVE:$' $0 | cut -d: -f1)
+cp /var/lib/docker/image/overlay2/repositories.json /tmp/orig.json
+echo "Extracting container images"
+tail -n +$((archive + 1)) $0 | openssl enc -pbkdf2 -d -aes256 -k $1 \\
+    | tar -C /var/lib/docker --strip-components=2 -x ./images
```

Comment:
Let's hope that a simple file-level merge of existing container images (dockerd's image and layer stores) will work (I've already forgotten details  on this matter :) )

Reply:
Will the archive itself stay on the filesystem after extraction?

Reply:
No, it won't. already verified.